korman/nodes/node_avatar.py

```diff
@@ -71,3 +71,388 @@ def export(self, exporter, bo, so):
+        col = layout.column()
+
+        col.prop(self, "forward")
+        col.prop(self, "backward")
+        col.prop(self, "stage_advance")
+        col.prop(self, "stage_regress")
+        col.separator()
+
+        col.label("Notify On:")
+        row = col.row()
+        row.prop(self, "notify_on")
+        col.separator()
+
+
+class PlasmaAnimStageSocket(PlasmaNodeSocketBase, bpy.types.NodeSocket):
+    bl_color = (0.188, 0.186, 0.349, 1.0)
+
+
+class PlasmaAnimStageNode(PlasmaNodeBase, bpy.types.Node):
+    bl_category = "AVATAR"
+    bl_idname = "PlasmaAnimStageNode"
+    bl_label = "Animation Stage"
+    bl_width_default = 325
+
+    pl_attrib = ("ptAttribAnimation")
+
+    anim_name = StringProperty(name="Animation Name",
+                               description="Name of animation to play")
+
+    loop_option = EnumProperty(name="Looping",
+                               description="Loop options for animation playback",
+                               items=[("kDontLoop", "Don't Loop", "Don't loop the animation"),
+                                      ("kLoop", "Loop", "Loop the animation a finite number of times"),
+                                      ("kLoopForever", "Loop Forever", "Continue playing animation indefinitely")],
+                               default="kDontLoop")
+    num_loops = IntProperty(name="Num Loops",
+                            description="Number of times to loop animation",
+                            default=0)
+
+    input_sockets = OrderedDict([
+        ("stage_settings", {
+            "text": "Stage Settings",
+            "type": "PlasmaAnimStageSettingsSocket",
+            "valid_link_nodes": "PlasmaAnimStageSettingsNode",
+            "valid_link_sockets": "PlasmaAnimStageSettingsSocket",
+            "link_limit": 1,
+        }),
+    ])
+
+    output_sockets = OrderedDict([
+        ("stage", {
+            "text": "Behavior",
+            "type": "PlasmaAnimStageRefSocket",
+            "valid_link_nodes": "PlasmaMultiStageBehaviorNode",
+            "valid_link_sockets": "PlasmaAnimStageRefSocket",
+        }),
+        ("stage_reference", {
+            "text": "Stage Progression",
+            "type": "PlasmaAnimStageOrderSocketOut",
+            "valid_link_nodes": "PlasmaAnimStageSettingsNode",
+            "valid_link_sockets": {"PlasmaAnimStageAdvanceSocketIn", "PlasmaAnimStageRegressSocketIn"} ,
+        }),
+    ])
+
+    def draw_buttons(self, context, layout):
+        col = layout.column()
+        col.prop(self, "anim_name")
+        col.separator()
+
+        row = col.row()
+        row.prop(self, "loop_option")
+        if self.loop_option == "kLoop":
+            row = col.row()
+            row.prop(self, "num_loops")
+        col.separator()
+
+    @property
+    def stage_id(self):
+        idx = None
+        if self.find_output_socket("stage").is_linked:
+            parent_MSB = self.find_output_socket("stage").links[0].to_node
```

Comment:
It would be a small performance win to only evaluate `self.find_output_socket("stage")` once.